### ISSUE TITLE
What jars must be included in a simple SWRL API application?

### ISSUE THREAD

**Issue by e-jajaga** [protegeproject/swrlapi#12]
I have a simple project for "playing" with OWL and SWRL APIs. I included OWL API required jars, but I am confuse about the SWRLAPI jars inclusion. I included swrlapi-1.0.0.jar and swrlapi-drools-engine-1.0.0.jar. I am running my sample code but receive the following exception:

Exception in thread "main" org.swrlapi.exceptions.SWRLRuleEngineException: Error creating rule engine Drools. Exception: java.lang.NoClassDefFoundError. Message: org/drools/runtime/rule/AgendaFilter
at org.swrlapi.factory.DefaultSWRLRuleAndQueryEngineFactory.createSWRLRuleEngine(DefaultSWRLRuleAndQueryEngineFactory.java:69)	at org.swrlapi.factory.DefaultSWRLRuleAndQueryEngineFactory.createSWRLRuleEngine(DefaultSWRLRuleAndQueryEngineFactory.java:69)
	at org.swrlapi.factory.DefaultSWRLRuleAndQueryEngineFactory.createSWRLRuleEngine(DefaultSWRLRuleAndQueryEngineFactory.java:45)
	at org.swrlapi.factory.SWRLAPIFactory.createSWRLRuleEngine(SWRLAPIFactory.java:80)
	at swrlapi5test.SWRLAPI5test.main(SWRLAPI5test.java:65)
Caused by: java.lang.NoClassDefFoundError: org/drools/runtime/rule/AgendaFilter
	at org.swrlapi.drools.core.DroolsSWRLRuleEngineCreator.create(DroolsSWRLRuleEngineCreator.java:27)
	at org.swrlapi.factory.DefaultSWRLRuleAndQueryEngineFactory.createSWRLRuleEngine(DefaultSWRLRuleAndQueryEngineFactory.java:61)
	... 3 more
Caused by: java.lang.ClassNotFoundException: org.drools.runtime.rule.AgendaFilter
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 5 more

**Comment by martinjoconnor** [comment]
You can find an example POM that builds a standalone application using the SWRLAPI and the Drools engine here:

https://github.com/protegeproject/swrltab/blob/master/pom.xml

It used the Maven assembly plugin to built a fat JAR containing all necessary libraries.

**Comment by e-jajaga** [comment]
Dear Martin,

Thank you for the support. The swrltab-1.0.0-jar-with-dependencies.jar works fine for my simple application. In a more complex scenario I need to integrate other jar(s) which like swrltab have also bindings of slf4j module. I tried to mvn build the swrltab adding the exception of slf4j module in the pom file like follows, but did not had a success:
```
<dependency>
      <groupId>edu.stanford.swrl</groupId>
      <artifactId>swrlapi-drools-engine</artifactId>
      <version>${swrlapi.version}</version>
	  <exclusions>
       <exclusion> 
          <groupId>org.slf4j</groupId>
          <artifactId>slf4j-log4j12</artifactId>
       </exclusion>
       <exclusion> 
          <groupId>log4j</groupId>
          <artifactId>log4j</artifactId>
      </exclusion>
    </exclusions> 
    </dependency>
<dependency>
      <groupId>org.slf4j</groupId>
      <artifactId>slf4j-log4j12</artifactId>
      <version>${slf4j.version}</version>
	  <exclusions>
       <exclusion> 
          <groupId>org.slf4j</groupId>
          <artifactId>slf4j-log4j12</artifactId>
       </exclusion>
       <exclusion> 
          <groupId>log4j</groupId>
          <artifactId>log4j</artifactId>
      </exclusion>
    </exclusions> 
</dependency>
```
How can I build a valid swrltab without the slf4j module?

**Comment by martinjoconnor** [comment]
I have cleaned up some of the SWRLAPI POMs and removed unused dependencies. The dependency on SLF4J is now gone.

**Comment by SmartR1** [comment]
Hello,
I am developing a java application under Eclipse where I am using an owl ontology. My objectif is to add SWRL rules from this application. For that, I integrated SWRLAPI . I downloaded the jars from this links 
https://jar-download.com/?detail_search=a%3A%22swrlapi-drools-engine%22&a=swrlapi-drools-engine

During the application run, an error is displayed:
java.lang.NoClassDefFoundError: org/swrlapi/factory/SWRLAPIFactory

I don't know if all necessary libraries are added to the Java classpath.
Note that I am not using Maven.

Thanks in advance
